src/UnitTests/Constructors.cs

```diff
@@ -5,6 +5,175 @@
 
 namespace AutoMapper.UnitTests.Constructors
 {
+    public class When_constructor_is_partial_match_with_value_type : AutoMapperSpecBase
+    {
+        GeoCoordinate _destination;
+
+        public class GeolocationDTO
+        {
+            public double Longitude { get; set; }
+            public double Latitude { get; set; }
+            public double? HorizontalAccuracy { get; set; }
+        }
+
+        public struct GeoCoordinate
+        {
+            public GeoCoordinate(double longitude, double latitude, double x)
+            {

```

Comment:
But if you meant if it were a class and we would validate constructors, then yes, it should :)